pkg/workload/schemachange/error_screening.go

```diff
@@ -79,36 +79,13 @@ func (og *operationGenerator) tableHasRows(
 func (og *operationGenerator) scanInt(
 	ctx context.Context, tx pgx.Tx, query string, args ...interface{},
 ) (i int, err error) {
-	err = tx.QueryRow(ctx, query, args...).Scan(&i)
-	if err == nil {
-		og.LogQueryResults(
-			query,
-			i,
-			args...,
-		)
-	}
-	return i, errors.Wrapf(err, "scanBool: %q %q", query, args)
+	return Scan[int](ctx, og, tx, query, args...)
 }
 
 func (og *operationGenerator) scanBool(
 	ctx context.Context, tx pgx.Tx, query string, args ...interface{},
 ) (b bool, err error) {
-	err = tx.QueryRow(ctx, query, args...).Scan(&b)
-	if err == nil {
-		og.LogQueryResults(
-			query,
-			b,
-			args...,
-		)
-	}
-	return b, errors.Wrapf(err, "scanBool: %q %q", query, args)
-}
-
-func scanString(
```

Comment:
Is there a reason we don't want a helper function for scanString along the lines of the above?

Basically
```
func scanString(...) {
    return Scan[string]
}
```?
I'm guessing its because its only used in one place currently?

Reply:
IMO a helper function of that nature isn't particularly useful. It's more or less the same amount of typing (`scanString` vs `Scan[string]`). It also creates a precedent that we should be making helper functions for each type, which is a net loss as we can just leverage generics.

I didn't eliminate the other helpers because they have a slightly different function signature and are used much more heavily. I can start to clean them up in a follow up commit?